### PR TITLE
270 inspection removal errors

### PR DESCRIPTION
**Pull request by Divinare**
Closes https://github.com/HSLdevcom/bultti/issues/270

3 Bug fixes here:
* ProcurementUnitMap = {} temporarily
* await Inspection.areaRequirements returned invalid areaRequirements
* setFrom didnt work, assigning properties 1 by 1

### CODE REVIEW

File: src/executionRequirement/plannedExecutionRequirement/areaRequirementUtils.ts [hunk 1]
```diff
@@ -19,9 +19,9 @@ import { PlannedAreaExecutionRequirement } from './PlannedAreaExecutionRequireme
 import { PlannedUnitExecutionRequirement } from './PlannedUnitExecutionRequirementEntity'
 import { PreInspection } from '../../inspection/PreInspectionEntity'
 
-// Execution requirement objects are saved in the DB only when the inspection is published.
+// Planned Area Execution requirement objects are saved in the DB only when the inspection is published.
 // Draft inspection requirements are calculated and returned without being saved.
-export async function getOrCreatePreInspectionRequirements(
+export async function getOrCreatePreInspectionAreaRequirements(
```
Comment (danieldunderfelt) [inline src/executionRequirement/plannedExecutionRequirement/areaRequirementUtils.ts:24]: It creates both unit and area requirements so this distinction is not needed.

File: src/executionRequirement/plannedExecutionRequirement/executionRequirementUtils.ts [hunk 2]
```diff
@@ -411,7 +411,12 @@ export async function removePreInspectionRequirements(
 
   let areaExecutionRequirements: PlannedAreaExecutionRequirement[] =
     (await inspection.areaExecutionRequirements) || []
-
+  // TODO: remove this filtering it works
+  // currently await inspection.areaExecutionRequirements returns values that have no id
+  // might be, for example due to a typeorm bug
+  areaExecutionRequirements = areaExecutionRequirements.filter(req => {
+    !!req.id
+  })
```
Comment (danieldunderfelt) [inline src/executionRequirement/plannedExecutionRequirement/executionRequirementUtils.ts:419]: I'll take a look at this today.

**Review by danieldunderfelt — CHANGES_REQUESTED** [review]
Some unnecessary changes should be reverted. I'll take a look at the problematic part today.

File: src/inspection/CommonInspectionResolver.ts [hunk 3]
```diff
@@ -262,6 +265,9 @@ export class CommonInspectionResolver
       throw new ApolloError('User lacks the permissions required to create this inspection.')
     }
 
+    if (!inspectionType) {
+      throw new ApolloError('InspectionType is missing.')
+    }
```
[the remark this answers is not in the document]
Comment (Divinare) [reply src/inspection/CommonInspectionResolver.ts:270]: Hmm I don't remember why I put this anymore. I guess it can be taken out then

**Comment by danieldunderfelt** [comment]
@Divinare Area requirement removal is now fixed!

**Comment by Divinare** [comment]
Tested removing an inspection with area requirements. Removing inspection removed area requirements so this can be merged now!

**Pull request merged**